src/main/java/com/blamejared/crafttweaker/api/recipes/ReplacementRule.java

```diff
@@ -0,0 +1,67 @@
+package com.blamejared.crafttweaker.api.recipes;
+
+import com.blamejared.crafttweaker.api.item.IIngredient;
+import net.minecraft.item.crafting.Ingredient;
+
+import java.util.Objects;
+
+// TODO("Make this more than a data class")
+public final class ReplacementRule {
```

Comment:
Do you have an idea on how a replacement rule could be exposed? Because currently, the generic type used may not be even exposed to ZenCode, nor is `Optional`